x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -248,7 +248,7 @@ public class SearchableSnapshots extends Plugin implements IndexStorePlugin, Eng
     public static final String DATA_TIERS_CACHE_INDEX_PREFERENCE = String.join(",", DataTier.DATA_CONTENT, DataTier.DATA_HOT);
     private static final int SEARCHABLE_SNAPSHOTS_INDEX_MAPPINGS_VERSION = 1;
 
-    private volatile Supplier<RepositoriesService> repositoriesServiceSupplier;
+    private final SetOnce<RepositoriesService> repositoriesService = new SetOnce<>();
```

Comment:
To confirm my understanding: this still needs to be a `SetOnce` only because we can't inject `RepositoriesService` as a constructor parameter of a plugin?

Reply:
Yep exactly. Once we can handle creation of Directory factories via injection, we wouldn't need to capture this anymore. In general, the uses of SetOnce in createComponents are because of a lack of injection or require parameters in other plugin interfaces.